.github/workflows/build-test-ship.yaml

```diff
@@ -131,21 +134,25 @@ jobs:
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v1
 
-      # On push/PR, ensure service images can be built on target platforms.
-      - name: Build final images
+      # Build and upload image artifact for use in api-proctor tests.
+      - name: Build image for api-proctor tests
         uses: docker/build-push-action@v2
-        if: |
-          github.event_name == 'pull_request' ||
-          (github.event_name == 'push' &&
-           steps.check-version-change.outputs.appchange == 'false')
```

Comment:
I've moved test builds (builds that just verify the service images can be built on the target platforms, without using the resulting images for anything) to a separate `build-for-target-platforms` job so they can run concurrently with the `build-and-ship-images` and `run-api-proctor-tests` jobs.